### PR TITLE
fix: Read environment variable configs in hello_triangle example

### PR DESCRIPTION
**Pull request by fryeb**
**Connections**
The hello_triangle doesn't support switching backends, leading to confusing errors like this one: https://github.com/gfx-rs/wgpu/issues/4247

**Description**
This adds support for the WGPU_* environment variables to the hello_triangle example, so that backend-specific issues can more quickly be narrowed down and fixes.

**Testing**
I've tested this manually, not sure if there is a better procedure for automatically testing these things

**Checklist**

- [x] Run `cargo fmt`.
- [x] Run `taplo format`.
- [x] Run `cargo clippy`. If applicable, add:
  - [x] `--target wasm32-unknown-unknown`
  - [ ] `--target wasm32-unknown-emscripten`
- [ ] Run `cargo xtask test` to run tests.
- [ ] Add change to `CHANGELOG.md`. See simple instructions inside file.

### CODE REVIEW

File: Cargo.lock
Comment (cwfitzgerald) [inline Cargo.lock:1]: Seemed to accidentally include some cargo.lock changes

**Review by cwfitzgerald — CHANGES_REQUESTED** [review]
LGTM, pending accidental .lock changes

(Reminder: please re-request a review from me once the changes are addressed to make sure I see it!)

**Comment by fryeb** [comment]
I've removed the accidentally-queued lock file stuff. Might be worth adding another utils method for getting the whole instance descriptor, but we'll see. Not worth blocking this quick fix imho

File: Cargo.lock
Comment (fryeb) [reply Cargo.lock:1]: oops! fixed this

**Comment by fryeb** [comment]
I had a spare couple of minutes so I factored the whole thing into the utilities package, hopefully that makes the examples a little more succinct.

**Review by cwfitzgerald — APPROVED** [review]
Now that's **chefs kiss**

**Comment by fryeb** [comment]
Apologies for re-requesting again. Just had to fix the clippy error

**Comment by cwfitzgerald** [comment]
No worries!

**Pull request merged**